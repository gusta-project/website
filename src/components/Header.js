import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { translate } from 'react-i18next';
import { Link } from 'gatsby-plugin-i18next';
import { Navbar, Nav } from 'react-bootstrap';

import LanguageSelector from './LanguageSelector';

class Header extends Component {
  static propTypes = {
    siteTitle: PropTypes.string.isRequired,
    t: PropTypes.func.isRequired
  };

  static defaultProps = {
    siteTitle: ''
  };

  render() {
    const { t } = this.props;

    return (
      <Navbar bg="dark" variant="dark" expand="lg" className="mb-5">
        <Navbar.Brand as={Link} to="/">
          {this.props.siteTitle}
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="mr-auto">
            <Nav.Link as={Link} to="/" activeClassName="active">
              {t('home')}
            </Nav.Link>
            <Nav.Link as={Link} to="/about" activeClassName="active">
              {t('about_us')}
            </Nav.Link>
            <Nav.Link
              href="https://github.com/mixnjuice"
              target="_blank"
              rel="noopener noreferrer"
            >
              GitHub
            </Nav.Link>
          </Nav>
          <LanguageSelector />
        </Navbar.Collapse>
      </Navbar>
    );
  }
}

export default translate()(Header);
