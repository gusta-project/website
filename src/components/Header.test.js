import React from 'react';
import renderer from 'react-test-renderer';

import Header from './Header';

jest.mock('components/LanguageSelector', () =>
  require('utils').mockComponent('LanguageSelector')
);

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Header siteTitle="Header Test" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
