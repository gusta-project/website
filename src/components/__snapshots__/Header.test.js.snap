// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<nav
  className="mb-5 navbar navbar-expand-lg navbar-dark bg-dark"
>
  <Link
    className="navbar-brand"
    to="/"
  >
    Header Test
  </Link>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <Link
        activeClassName="active"
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        onClick={[Function]}
        to="/"
      >
        home
      </Link>
      <Link
        activeClassName="active"
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        onClick={[Function]}
        to="/about"
      >
        about_us
      </Link>
      <a
        className="nav-link"
        data-rb-event-key="https://github.com/gusta-project"
        href="https://github.com/gusta-project"
        onClick={[Function]}
        onKeyDown={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
    </div>
    <LanguageSelector />
  </div>
</nav>
`;
